bastion/src/path.rs

```diff
@@ -0,0 +1,459 @@
+                Some(BastionPathElement::Supervisor(..)) => {
+                    write!(f, "Child is not appendable to a supervisor")
+                }
+                Some(BastionPathElement::Children(..)) => unreachable!(),
+                Some(BastionPathElement::Child(..)) => {
+                    write!(f, "Child is not appendable to a child")
+                }
+            },
+        }
+    }
+}
+
+impl fmt::Debug for AppendError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "Can't append {:?} to {:?}", self.element, self.path)
+    }
+}
+
+impl BastionPath {
+    pub(crate) fn append(self, el: BastionPathElement) -> Result<BastionPath, AppendError> {
+        match el {
+            sv @ BastionPathElement::Supervisor(_) => match self.this {
+                None => Ok(BastionPath {
+                    parent_chain: self.parent_chain,
+                    this: Some(sv),
+                }),
+                Some(BastionPathElement::Supervisor(id)) => {
+                    let mut path = BastionPath {
+                        parent_chain: self.parent_chain,
+                        this: Some(sv),
+                    };
+                    path.parent_chain.push(id);
+                    Ok(path)
+                }
+                this => Err(AppendError {
+                    path: BastionPath {
+                        parent_chain: self.parent_chain,
+                        this,
+                    },
+                    element: sv,
+                }),
+            },
+            children @ BastionPathElement::Children(_) => match self.this {
+                Some(BastionPathElement::Supervisor(id)) => {
+                    let mut path = BastionPath {
+                        parent_chain: self.parent_chain,
+                        this: Some(children),
+                    };
+                    path.parent_chain.push(id);
+                    Ok(path)
+                }
+                this => Err(AppendError {
+                    path: BastionPath {
+                        parent_chain: self.parent_chain,
+                        this,
+                    },
+                    element: children,
+                }),
+            },
+            child @ BastionPathElement::Child(_) => match self.this {
+                Some(BastionPathElement::Children(id)) => {
+                    let mut path = BastionPath {
+                        parent_chain: self.parent_chain,
+                        this: Some(child),
+                    };
+                    path.parent_chain.push(id);
+                    Ok(path)
+                }
+                this => Err(AppendError {
+                    path: BastionPath {
+                        parent_chain: self.parent_chain,
+                        this,
+                    },
+                    element: child,
+                }),
+            },
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
Let's move these tests to `tests` directory.

Reply:
I can't move them because this test uses non-public API

Reply:
Ah ok. no problem.